tests/e2e/redis/basic-ingest-redis-serving.py

```diff
@@ -299,6 +299,157 @@ def try_get_features():
+            "customer2_past_transactions_string": [['first_'+str(i),'second_'+str(i)] for i in range(N_ROWS)],
+            "customer2_past_transactions_bool": [[True,False] for _ in range(N_ROWS)]
+        }
+    )
+    return customer_df
+
+@pytest.fixture(scope='module')
+def entity_list_type_dataframe():
+    N_ROWS = 2
+    time_offset = datetime.utcnow().replace(tzinfo=pytz.utc)
+    customer_df = pd.DataFrame(
+        {
+            "datetime": [time_offset] * N_ROWS,
+            "district_ids": [[np.int64(i),np.int64(i+1),np.int64(i+2)] for i in range(N_ROWS)],
+            "district_rating": [i for i in range(N_ROWS)],
+            "district_cost": [float(i)+0.5 for i in range(N_ROWS)],
+            "district_past_transactions_int": [[i,i+2] for i in range(N_ROWS)],
+            "district_past_transactions_double": [[float(i)+0.5,float(i)+2] for i in range(N_ROWS)],
+            "district_past_transactions_float": [[float(i)+0.5,float(i)+2] for i in range(N_ROWS)],
+            "district_past_transactions_string": [['first_'+str(i),'second_'+str(i)] for i in range(N_ROWS)],
+            "district_past_transactions_bool": [[True,False] for _ in range(N_ROWS)]
+        }
+    )
+    return customer_df
+
+
+@pytest.mark.timeout(600)
+@pytest.mark.run(order=14)
+def test_basic_retrieve_online_dict(client, list_type_dataframe, entity_list_type_dataframe):
+    # Case 1: Multiple entities check
+    customer_fs = FeatureSet(
+        name="customer2",
+        features=[
+            Feature(name="customer2_rating", dtype=ValueType.INT64),
+            Feature(name="customer2_cost", dtype=ValueType.FLOAT),
+            Feature(name="customer2_past_transactions_int", dtype=ValueType.INT64_LIST),
+            Feature(name="customer2_past_transactions_double", dtype=ValueType.DOUBLE_LIST),
+            Feature(name="customer2_past_transactions_float", dtype=ValueType.FLOAT_LIST),
+            Feature(name="customer2_past_transactions_string", dtype=ValueType.STRING_LIST),
+            Feature(name="customer2_past_transactions_bool", dtype=ValueType.BOOL_LIST)
+        ],
+        entities=[Entity("customer_id2", ValueType.INT64)],
+        max_age=Duration(seconds=3600)
+    )
+
+    client.set_project(PROJECT_NAME)
+    client.apply(customer_fs)
+
+    customer_fs = client.get_feature_set(name="customer2")
+    client.ingest(customer_fs, list_type_dataframe, timeout=600)
+    time.sleep(15)
+
+    online_request_entity = [{"customer_id2": 0},{"customer_id2": 1}]
+    online_request_features = [
+        "customer2_rating",
+        "customer2_cost",
+        "customer2_past_transactions_int",
+        "customer2_past_transactions_double",
+        "customer2_past_transactions_float",
+        "customer2_past_transactions_string",
+        "customer2_past_transactions_bool"
+    ]
+
+    online_features_actual = client.get_online_features(entity_rows=online_request_entity, feature_refs=online_request_features)
```

Comment:
Unfortunately, I think you need to poll `get_online_features` for feature values as they might not be immediately available (even after your `time.sleep`). See [here](https://github.com/feast-dev/feast/blob/b461f4dea517de5e5ad7c9114e069081d9437810/tests/e2e/redis/basic-ingest-redis-serving.py#L261) for how to do it. Same for `online_features_actual2`.